batch/batch/driver/instance_collection/job_private.py

```diff
@@ -402,11 +426,20 @@ async def create_instance_with_error_handling(
                         batch_format_version = BatchFormatVersion(record['format_version'])
                         spec = json.loads(record['spec'])
                         machine_spec = batch_format_version.get_spec_machine_spec(spec)
-                        instance, total_resources_on_instance = await self.create_instance(machine_spec)
+
+                        regions = json_to_value(record['regions'])
+                        if regions is None:
+                            regions = self.inst_coll_manager.regions
+
+                        instance, total_resources_on_instance = await self.create_instance(machine_spec, regions)
                         log.info(f'created {instance} for {(batch_id, job_id)}')
                         await mark_job_creating(
                             self.app, batch_id, job_id, attempt_id, instance, time_msecs(), total_resources_on_instance
                         )
+                    except RegionsNotSupportedError:
+                        await mark_job_errored(
+                            self.app, batch_id, job_id, attempt_id, record['user'], record['format_version']
```

Comment:
why is there no message included in this mark_job_errored?

Reply:
that's a mistake. will fix it.